### PR TITLE
SingleSelect: Clean Up and Bug Fix

### PR DESCRIPTION
**Pull request by pfafffabian-ifx**
By creating this pull request you agree to the terms in CONTRIBUTING.md.
https://github.com/Infineon/.github/blob/master/CONTRIBUTING.md
--- DO NOT DELETE ANYTHING ABOVE THIS LINE ---

CONTRIBUTING.md also tells you what to expect in the PR process.

Description
- Cleanup of the Single Select codebase
- Fix Bug where the disabled state of the Single Select did not work as expected

Related Issue
- #1413
- #1328

<details>
  <summary>📦 Published PR as canary version: <code>24.6.0--canary.1414.a6c43d47620526637e1e4b3ca1a9529a40289ba5.0</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @infineon/infineon-design-system-stencil@24.6.0--canary.1414.a6c43d47620526637e1e4b3ca1a9529a40289ba5.0
  # or 
  yarn add @infineon/infineon-design-system-stencil@24.6.0--canary.1414.a6c43d47620526637e1e4b3ca1a9529a40289ba5.0
  ```
</details>

### CODE REVIEW

File: packages/components/src/components/select/single-select/select.scss [hunk 1]
```diff
@@ -79,6 +79,17 @@
       -webkit-user-select: none;
       -ms-user-select: none;
       user-select: none;
+
+      div {
```
Comment (tishoyanchev) [inline packages/components/src/components/select/single-select/select.scss:83]: Which `div` are you styling? Please add a class, and don't apply styles to all `<div>`s.

**Comment by tishoyanchev** [comment]
@pfafffabian-ifx 
- `options` prop should not be mutated. If creating an internal state for this Prop is not feasible, then at least make it mutable. 
- still getting the `selectedOption`'s warning in the console but only in vanilla. Remove `this.selectedOption = null;` from  `private init() {..}`. This solves the issue. Make sure it's not needed to have it first. 
- remove `no-allow cursor` on `disabled` for single and multi select. When disabled, mouse cursor should be `default`.
- when `disabled`, middle area is still clickable

File: packages/components/src/components/select/single-select/select.scss (hunk 1 above)
Comment (pfafffabian-ifx) [reply packages/components/src/components/select/single-select/select.scss:83]: The problem regarding the style was this in the react example (App.scss):
```scss
div {
  background-color: tokens.$ifxColorBaseWhite;
  // --ifx-font-family: 'Arial black'; //overwrite default font
}
```

I wasn't sure whether I should remove it in the example, or override it in the component, but I'll change it.

**Comment by pfafffabian-ifx** [comment]
* `options` prop should not be mutated. If creating an internal state for this Prop is not feasible, then at least make it mutable. 
  * I marked it as mutable, couldn't get it to run otherwise
* still getting the `selectedOption`'s warning in the console but only in vanilla. Remove `this.selectedOption = null;` from  `private init() {..}`. This solves the issue. Make sure it's not needed to have it first. 
  * I removed it. It does not seem to be required
* remove `no-allow cursor` on `disabled` for single and multi select. When disabled, mouse cursor should be `default`.
  * Fixed
* when `disabled`, middle area is still clickable
  * I tested it in every example app. For me it isn't clickable

**Comment by tishoyanchev** [comment]
@pfafffabian-ifx 
- on multi-select, the no-allow cursor remains on disabled
- on single select, when disabled, the central area has the 'pointer' cursor, and it's clickable, and you can open the menu.

**Comment by verena-ifx** [comment]
I added the disabled check for the clearSelection button and made sure the property naming in the stories file is consistent to the rest (currently we just use the props, not different names - in caps - for them). 
I also merged the latest changes into the branch.

**Review by verena-ifx — APPROVED** [review]

**Pull request merged**